fix: Reset active app loader after service update during service update

keulinho opened the pull request:
fixes https://github.com/shopware/shopware/issues/11515


### 1. Why is this change necessary?
during update process the swag insider preview app is deleted, but the asset install command relies on a old state, because the active apps loader is not manually resettet, therefore it tries to install assets from an app that was deleted

### 2. What does this change do, exactly?
reset the state of the active apps loader after services were synced

### 3. Describe each step to reproduce the issue or behaviour.
Update from 6.7.0.0 with insider previews available to latest 6.7.1.0

### 4. Please link to the relevant issues (if any).
fixes https://github.com/shopware/shopware/issues/11515

keulinho committed:
fix: Reset active app loader after service update during service update

fixes https://github.com/shopware/shopware/issues/11515

Gaitholabi reviewed (commented):
I would move this call to inside `removeOrphanedServices` - to `Shopware\Core\Service\LifecycleManager`, but otherwise looks good 🤔

keulinho committed:
Use app state events to reset app loader internal cache

keulinho commented:
@Gaitholabi i thought about the same, but actually then you likely have to add it to every other call in there that might add or remove services

I now settled to add event listener to app activated and app deactivated events to call `reset()` that way the root cause should be fixed and it shouldn't happen anymore that the active apps loader returns outdated data

keulinho committed:
Merge branch 'trunk' into fix/reset-active-app-loader

keulinho committed:
Reset active apps loader after deactivating app

keulinho commented:
ok turns out the `AppStateService`already called the `reset()` method manually on app activated and deactivated.

However for deactivation we called it before we dispatched the deactivated event and before we changed the state in the DB, this lead to the issue, because during AppDeactivated event we have a listener that rebuilds the theme if necessary and there we access the ActiveAppsLoader again, and because that is executed before the state is changed in the DB the old app is again in the cached state, therefore the fix was to move the reset call to the bottom of the method after we changed the status of the app in the DB, so that the old app won't show up in the active apps loader again

frobel commented:
Update from 6.6.10.x to your branch :heavy_check_mark: 
Update from 6.7.0.1 to your branch :heavy_check_mark: 
In both cases services are installed :heavy_check_mark: 
SwagInsiderPreview is not available anymore :heavy_check_mark:
